33 synchronous konserve io

whilo opened the pull request:
Use new konserve synchronous interface and remove blocking `core.async` operators.

whilo committed:
Use new konserve synchronous interface.

Removes core.async from callstack in synchronous case.

whilo committed:
Bump konserve version.

whilo committed:
Add configuration flag to deactivate crypto-hashing.

whilo committed:
Fix tests.

whilo committed:
Fix format.

whilo committed:
Bump konserve.

grischoun commented on test/hitchhiker/konserve_test.cljc:
```diff
@@ -151,4 +150,4 @@
                                     num-ops)]
-                   (ha/<?? (ops-test ops universe-size)))))
\ No newline at end of file
+                   (ha/<?? (ops-test ops universe-size)))))
```
I can't see who calls this function. So is this ever run?

whilo replied:
Good catch. It is still tested in `ClojureScript` with the recorded ops, but not here. I am not sure, but i think there was some problem with this test, but we should definitely run it again. @kordano Can you try whether you can call this function and create a test similar to the cljs one?

grischoun reviewed (commented):
It looks ok. The test I could run works.
Though honestly I can't say much about the full PR. There are still many things I do not understand. 
E.g.,  what does a `KonserveAdr` represent?
I also wonder what `hash-node?` means?

whilo commented:
`KonserveAddress` is a pointer to the underlying storage that allows the data (`IndexNode` or `DataNode`) to be loaded when it is needed while walking an iterator. `hash-node?` is a new configuration parameter that I have introduced to decide whether `hasch.core/uuid` is used to create a cryptographic hash on the value or whether to just use a random uuid (which is faster, but does not guarantee integrity [which does not matter yet, but will be crucial once snapshots are distributed and signed]).

whilo committed:
Add versioning, remove random tags from ops.

Tree generation is now deterministic and yields the same crypto hashes. Also
deactivate cljs tests for now until konserve is updated.

whilo committed:
Fix format.

kordano commented:
superseded by #35 since we moved to the single release branch approach.
